### ISSUE TITLE
Failed to get MParticle instance, getInstance() called prior to start() on any version greater than 5.14.16

### ISSUE THREAD

**Issue by jcesarmobile** [mParticle/mparticle-android-sdk#66]
I'm using mParticle in a webview like this:

```
MParticleOptions options = MParticleOptions.builder(this.getContext())
                .credentials("mykey", "mysecret")
                .environment(MParticle.Environment.AutoDetect)
                .logLevel(MParticle.LogLevel.NONE)
                .build();
MParticle.start(options);
MParticle.getInstance().registerWebView(this.getWebView());
```

As you can see, I'm calling start before getInstance, but I get the `Failed to get MParticle instance, getInstance() called prior to start()` error.
The error comes from start itself, is called before my call to getInstance, but from the start own internal getInstance call.

On 5.14.16 I don't get the error, but I get it on any newer version, starting 5.15.0, including latest (5.17.1 at the moment)

**Comment by alexs-mparticle** [comment]
@jcesarmobile Thanks for opening the issue. I'll bring this up to the team and we'll see if we can get you an answer as soon as we can.

**Comment by alexs-mparticle** [comment]
Tracked internally as 67081

**Comment by willpassidomo** [comment]
Hey @jcesarmobile I just want to double-check, are you seeing a log error or an Exception? If it is the latter, that is a big problem and we'll work to fix that asap. If it is the former, I believe this started as a result of some additional null-checks we added in a recent version.

We've had our eye on this issue for a while, but essentially when you call `MParticle.getInstance()` from client code or our own internal code, we will log the error message you observe if the instance is `null`. This has become a nuisance as our codebase has grown since we have some instances where the SDK could legitimately be started or not, so every time we call `if (MParticle.getInstance() != null)` or `MParticle.getInstance()?.` the log message will print, regardless of whether we might have been expecting it to be `null` or not. We have a fix for this in our backlog, but I think with the recent increase in the volume of these logs with those additional null-checks, we should increase the priority

**Comment by jcesarmobile** [comment]
It's an error message on logcat

**Comment by willpassidomo** [comment]
Ok, sounds good. I've opened a ticket to clean up some of these unnecessary log messages. I appreciate you taking the time to report this and please let me know if anything else comes up :)

**Comment by florianmski** [comment]
Hello, sorry to re-open this issue but I'm just starting to implement mParticle in our app and I see the same log. I'm literally only calling `start` for now and already spent quite some time trying to debug this error message but from what I'm seeing here this seems to be expected in some case?
Cheers

**Comment by willpassidomo** [comment]
Hey @florianmski, sorry this is causing problems again. It looks like this is a similar case to what I described in my previous comment, our internal code is checking if the singleton has been set and triggering the error log. Since this log doesn't indicate an actual error, I am going to either downward revise the log level to `info` or something similar or remove it altogether.

**Comment by florianmski** [comment]
@willpassidomo thanks for the quick response, this sounds good to me

**Comment by willpassidomo** [comment]
released in version 5.36.0
